app/services/concerns/monthly_equivalent_calculatable.rb

```diff
@@ -1,21 +1,14 @@
 module MonthlyEquivalentCalculatable
 private
 
-  def determine_calc_method(period)
-    case period.to_sym
-    when :monthly
-      :monthly_to_monthly
-    when :four_weekly
-      :four_weekly_to_monthly
-    when :two_weekly
-      :two_weekly_to_monthly
-    when :weekly
-      :weekly_to_monthly
-    when :unknown
-      :blunt_average
-    else
-      raise ArgumentError, "unexpected period #{period}"
-    end
+  def determine_calc_method(frequency)
+    raise ArgumentError, "unexpected frequency #{frequency}" unless frequency_conversions.key?(frequency)
+
+    frequency_conversions[frequency]
+  end
+
+  def three_monthly_to_monthly(value)
+    (value / 3).round(2)
```

Comment:
I do find it concerning that we're doing floating point arithmetic with monetary values —particularly given the potential negative impact a variation could have on someone's eligibility result.

That is probably worth a wider discussion though, because it's prevalent throughout several of the services.